django/contrib/admin/options.py

```diff
@@ -606,13 +612,14 @@ def get_form(self, request, obj=None, **kwargs):
             fields = kwargs.pop('fields')
         else:
             fields = flatten_fieldsets(self.get_fieldsets(request, obj))
-        if self.exclude is None:
+        if self.get_exclude(request, obj) is None:
             exclude = []
         else:
-            exclude = list(self.exclude)
+            exclude = list(self.get_exclude(request, obj))
```

Comment:
I think you missed the comment on the previous PR, "I don't know if the `get_exclude()` method might be expensive, but it seems we could simplify by storing the result in a local variable to avoid calling it so many times."